app/src/main/java/fr/free/nrw/commons/contributions/MainActivity.java

```diff
@@ -111,6 +113,16 @@ protected void onSaveInstanceState(Bundle outState) {
         outState.putInt("viewPagerCurrentItem", viewPager.getCurrentItem());
     }
 
+    @Override
+    public void requestMoreImages() {
+
+    }
+
+    @Override
+    public void viewPagerNotifyDataSetChanged() {
+
+    }
+
     private void initMain() {
```

Comment:
Please remove the indentation changes, also instead of making the MainActivity implement this interface, should we consider adding a check in MediaDetails that if getContext is an instance of CategoryImagesCallback, then and only then perform the cast and do the necessary operations. That would leave the MainActivity untouched, what do you think?

Reply:
@ashishkumar468 done :)

Reply:
@kbhardwaj123  Did you read the entire comment ? :P

Reply:
@ashishkumar468 oops :), i think thats a better idea i will test that

Reply:
@ashishkumar468 but there's a problem with that i mean if getContext is not an instanceof CategoryImagesCallback then i wont't typecast it and try to access `requestMoreImages()` it would again result in a crash since MainActivity won't have `requestMoreImages()` method

Reply:
I meant call requesMoreImages only if the context is an instanceOf CategoryImagesCallback
```
if (i+1 >= adapter.getCount() && getContext() instanceOf CategoryImagesCallback){
            ((CategoryImagesCallback) getContext()).requestMoreImages();
}
```

Reply:
@ashishkumar468 oh sorry i should have caught that a little earlier